pkg/util/iptables/iptables_test.go

```diff
@@ -934,7 +938,7 @@ func TestRestoreAll(t *testing.T) {
 			func(cmd string, args ...string) exec.Cmd { return fakeexec.InitFakeCmd(&fcmd, cmd, args...) },
 		},
 	}
-	runner := newInternal(&fexec, ProtocolIPv4, TestLockfilePath)
+	runner := newInternal(&fexec, ProtocolIPv4, getLockPath14x(), TestLockfilePath)
```

Comment:
it seems random to have a hardcoded testing lock name for 1.6 but a per-process generated one for 1.4...

Reply:
According to the report of previous tests https://testgrid.k8s.io/sig-release-master-blocking#bazel-test-master&width=20&sort-by-name=&include-filter-by-regex=pkg/util/iptables, looks like only acquring lock for 1.4 would time out, I am a bit curious as well.

Reply:
> it seems random to have a hardcoded testing lock name for 1.6 but a per-process generated one for 1.4...

I agree

> looks like only acquring lock for 1.4 would time out, I am a bit curious as well.

TestLockfilePath is not fully-qualified, perhaps the working dir is different per test-run? I agree that we should be consistent in how we assign random filenames for the test

Reply:
> TestLockfilePath is not fully-qualified

The original 1.4 lock file `@xtables` is neither fully-qualified.

> I agree that we should be consistent in how we assign random filenames for the test

I am okay to update the tests.

Reply:
> The original 1.4 lock file `@xtables` is neither fully-qualified.

abstract sockets are handled differently than lock files and are not treated as pathnames

Reply:
> abstract sockets are handled differently than lock files and are not treated as pathnames

just out of curiosity, I am sorry if I ask a dumb question. Perhaps I am missing something, currently `net.ListenUnix("unix", &net.UnixAddr{Name: "@xtables", Net: "unix"})` is called to grab the lock file https://github.com/kubernetes/kubernetes/blob/0e58014d71ba92a055e41541b3b7b7373cc0343e/pkg/util/iptables/iptables_linux.go#L84-L89
If there is no error, an unix domain socket named `@xtables` will be created under the curret working dir. I am confused why `@xtables` is not treated as pathname?

Reply:
> If there is no error, an unix domain socket named `@xtables` will be created under the curret working dir.

Only on OSX. On linux, `@xtables` is an abstract socket in a global namespace unrelated to filesystem paths.

Reply:
Ahhh got it! Thanks for your explanation!